ecs_client_test.go

```diff
@@ -103,3 +280,11 @@ func checkAWSEnvVars(t *testing.T) {
 		assert.FailNow(t, fmt.Sprintf("missing required AWS environment variables: %s", missing))
 	}
 }
+
+func makeFamily(name string) string {
+	test := strings.Split(name, "/")
+	if len(test) > 1 {
+		return fmt.Sprintf("%v-%v-%v-%v-%v", os.Getenv("AWS_ECS_TASK_DEFINITION_PREFIX"), "cocoa", test[0], test[1], utility.RandomString())
+	}
+	return fmt.Sprintf("%v-%v-%v-%v", os.Getenv("AWS_ECS_TASK_DEFINITION_PREFIX"), "cocoa", test[0], utility.RandomString())
```

Comment:
Use `%s`. Default to using `%s` if we already know all the types are strings (or can be stringified because they implement `String()`). `%v` is typically used for more verbose formatting, but is unnecessary for strings.

Can we use `strings.Join` instead of `fmt.Sprintf`? It's easier to join an arbitrary number of strings by the same delimiter (`-` in this case) if you use `strings.Join` rather than `fmt.Sprintf`. It's also easier to add/remove strings later on, since you don't have to remember to update the formatting string with the correct number of `%s` formatters.

Reply:
Makes sense, I'll use `strings.Join`